pkg/operator/operator_test.go

```diff
@@ -89,100 +136,45 @@ func TestOperatorSync_NoOp(t *testing.T) {
 		t.Fatalf("could not write the images file: %v", err)
 	}
 
+	optrConfig := newOperatorConfig()
+	d := newDeployment(optrConfig, map[string]bool{})
+
 	for _, tc := range cases {
 		t.Run(string(tc.platform), func(t *testing.T) {
-			infra := &configv1.Infrastructure{
+			infra := &v1.Infrastructure{
 				ObjectMeta: metav1.ObjectMeta{
 					Name: "cluster",
 				},
-				Status: configv1.InfrastructureStatus{
+				Status: v1.InfrastructureStatus{
 					Platform: tc.platform,
 				},
 			}
 
-			configClient := fakeos.NewSimpleClientset()
-			configSharedInformer := configinformersv1.NewSharedInformerFactoryWithOptions(configClient, 10*time.Minute)
-			featureGateInformer := configSharedInformer.Config().V1().FeatureGates()
-
-			optr := Operator{
-				eventRecorder:     record.NewFakeRecorder(5),
-				osClient:          fakeconfigclientset.NewSimpleClientset(infra),
-				imagesFile:        imagesFilePath,
-				featureGateLister: featureGateInformer.Lister(),
-			}
+			stopCh := make(<-chan struct{})
+			optr := newFakeOperator([]runtime.Object{d}, []runtime.Object{infra}, stopCh)
```

Comment:
why are we passing the deployment here? shouldn't it be created by the fake client on sync?
Also can we validate for noop providers is not created?

Reply:
done